app/controllers/users_controller.rb

```diff
@@ -21,6 +29,30 @@ def create
     end
   end
 
+  def edit
+    return if @user
+    flash[:warning] = t :show
+    redirect_to root_path
```

Comment:
remove the body of the edit method as you are doing the same while loading user in the before_action filter